osu.Game/GameModes/Play/BeatmapGroup.cs

```diff
@@ -0,0 +1,246 @@
+//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Beatmaps;
+using osu.Game.Database;
+using osu.Framework.Graphics.Primitives;
+using OpenTK;
+using System.Linq;
+using osu.Framework.Graphics.Transformations;
+using osu.Framework.Input;
+using OpenTK.Graphics;
+using osu.Game.Beatmaps.IO;
+using osu.Framework.Graphics.Textures;
+using System.Threading.Tasks;
+
+namespace osu.Game.GameModes.Play
+{
+    class BeatmapGroup : Container
+    {
+        private const float collapsedAlpha = 0.5f;
+        private const float collapsedWidth = 0.8f;
+        
+        private BeatmapInfo selectedBeatmap;
+        public BeatmapInfo SelectedBeatmap
+        {
+            get { return selectedBeatmap; }
+            set
+            {
+                selectedBeatmap = value;
+            }
+        }
+
+        public event Action<BeatmapSetInfo> SetSelected;
+        public event Action<BeatmapSetInfo, BeatmapInfo> BeatmapSelected;
+        public BeatmapSetInfo BeatmapSet;
+        private BeatmapSetBox setBox;
+        private FlowContainer topContainer;
+        private FlowContainer difficulties;
+        private bool collapsed;
+        public bool Collapsed
+        {
+            get { return collapsed; }
+            set
+            {
+                if (collapsed == value)
+                    return;
+                collapsed = value;
+                this.ClearTransformations();
+                const float uncollapsedAlpha = 1;
+                Transforms.Add(new TransformAlpha(Clock)
+                {
+                    StartValue = collapsed ? uncollapsedAlpha : collapsedAlpha,
+                    EndValue = collapsed ? collapsedAlpha : uncollapsedAlpha,
+                    StartTime = Time,
+                    EndTime = Time + 250,
+                });
+                if (collapsed)
+                    topContainer.Remove(difficulties);
+                else
+                    topContainer.Add(difficulties);
+                setBox.ClearTransformations();
+                setBox.Width = collapsed ? collapsedWidth : 1; // TODO: Transform
+                setBox.BorderColour = new Color4(
+                    setBox.BorderColour.R,
+                    setBox.BorderColour.G,
+                    setBox.BorderColour.B,
+                    collapsed ? 0 : 255);
+                setBox.GlowRadius = collapsed ? 0 : 5;
+            }
+        }
+        
+        private void updateSelected(BeatmapInfo map)
```

Comment:
also i feel like this method should go inside the property setter rather than be on its own.